docs/guides/implementation_blueprint/data-synchronization.md

```diff
@@ -0,0 +1,200 @@
+
+> 🚧 Customer data synchronization and `source_id`
+>
+> If your data is to be synchronized based on the `source_id` of the customer, you need to import the `source_id` when the customer is uploaded for the first time. You will not be able to change or update `source_id` later on.
+
+### Customer synchronization options
+
+<!-- https://success.voucherify.io/article/430-crm-playbook
+Adding customers
+
+These are the ways you can add customer profiles to Voucherify:
+
+API
+Webhooks
+The import with a CSV file. 
+You can integrate your CRM platform with Voucherify.
+
+ -->
+
+#### Upserting customers
+
+<!-- do we clarify what `upsert` means: `update` + `insert`? -->
+
+If the customer does not exist in Voucherify, they are created automatically in a request that includes the customer object. If the customer exists, their data will be upserted. In this case, the response also includes the `updated_at` object. <!-- these are objects, right? -->
+
+The following actions <!-- or activities? --> upsert customer data:
+- order creation
+- voucher publication
+- validation <!-- IMO should be here-->
+- redemption
+
+<!-- Redemption... but this is to be deprecated, right? so should we actually link/mention those
+
+I need clarification – we want customers to use qualification endpoint or the stackable discounts endpoint, and these do redeem, so is this info `redemption` all right?
+
+Anything else that MUST be here? I don't mean everything
+
+-->
+
+<!-- ##### Order creation
+
+##### Voucher publication
+
+##### Redemption
+
+-->
+
+#### Importing customers
+
+You can import your customer database to Voucherify with a CSV file. If you use a CRM system, use its unique ID as a `source_id` field.
+
+Read our [customer import guide](https://support.voucherify.io/article/67-how-to-import-my-customers#csv-import "How to import my customers?") to learn more.
+
+<!-- Should I write a full step-by-step manual for this? -->
+
+#### Creating and updating customers via the API
+
+If you want to keep your customers database up to date with Voucherify, create a customer in Voucherify every time a new user is added to your database. Use the [Create customer](ref:create-customer) API endpoint to create customer data.
+
+To update customer data, use [Update customer](ref:update-customer) API endpoint. The [Create customer](ref:create-customer) API endpoint can be also used to update customer data.
+
+<!-- Simply repeat the endpoints? -->
+
+#### Synchronizing with connectors
+
+Customer data can be upserted with integrated platforms that support outbound traffic:
+- Batch
+- Bloomreach Engagement
+- Intercom
+- Klaviyo
+- MoEngage
+- mParticle
+- Salesforce
+- Segment
+- ZOHO
+
+<!-- Again, clarification: do we want to add only those platforms that send customer 
+data to Voucherify, right? 
+
+I'll update the links once these get clarified
+
+Also: how to know which platform sends data to V% and which receives V% data?
```

Comment:
I needed to review the documentation to get that info. It's not that straightforward since it depends on the implementation and the context of using the data. If the data consists of customer sync, then it's fine. If the solution allows sending any API POST request (Braze/Bloomreach Engagement), it can also be used for syncing.

Reply:
Ok, I think I get it 🙂